### PR TITLE
Refactor Article and Page with concerns

### LINKED ISSUE #72: Refactor Page/Article models
- DRY
- import a Post module

### PR DESCRIPTION
**Pull request by guitsaru**
Also added some concerns for some other heavily duplicated code: slug generation and name from title and subtitle.

Closes #72

### CODE REVIEW

File: app/models/concerns/post.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+module Post
+  extend ActiveSupport::Concern
+
+  include NameFromTitle
+  include Slug
```
Comment (veganstraightedge) [inline app/models/concerns/post.rb]: My only small issue with this is the naming. It feels inconsistent.

Feels like it should be either:
- `Name` and `Slug` or
- `NameFromTitle` (though technically it's from title and subtitle) and `SlugX` (not sure what X would be though)

Is there a pattern or naming convention for things like this?

Comment (guitsaru) [inline app/models/concerns/post.rb]: Yeah.  I started with just `Name`, but changed it when lots of models implemented name differently.  I'm fine with changing it back, but wanted to make sure it wasn't confusing.

Comment (guitsaru) [inline app/models/concerns/post.rb]: Maybe `Nameable` and `Sluggable` are closer to the actual convention, but then `Post` doesn't quite fit.

Comment (guitsaru) [inline app/models/concerns/post.rb]: I've changed it back to `Name` for now.  I think that's the simplest option, but it's easy to change later.

**Pull request merged**